src/library-authoring/component-info/ComponentAdvancedAssets.tsx

```diff
@@ -0,0 +1,95 @@
+/* eslint-disable no-nested-ternary */
+/* eslint-disable import/prefer-default-export */
+import React from 'react';
+import {
+  Button,
+  Dropzone,
+} from '@openedx/paragon';
+import { Delete } from '@openedx/paragon/icons';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import { FormattedMessage, FormattedNumber, useIntl } from '@edx/frontend-platform/i18n';
+
+import { LoadingSpinner } from '../../generic/Loading';
+import DeleteModal from '../../generic/delete-modal/DeleteModal';
+import { useLibraryContext } from '../common/context';
+import { getXBlockAssetsApiUrl } from '../data/api';
+import { useDeleteXBlockAsset, useInvalidateXBlockAssets, useXBlockAssets } from '../data/apiHooks';
+import messages from './messages';
+
+export const ComponentAdvancedAssets: React.FC<Record<never, never>> = () => {
+  const intl = useIntl();
+  const { readOnly, sidebarComponentInfo } = useLibraryContext();
+
+  const usageKey = sidebarComponentInfo?.id;
+  // istanbul ignore if: this should never happen in production
+  if (!usageKey) {
+    throw new Error('sidebarComponentUsageKey is required to render ComponentAdvancedAssets');
+  }
+
+  // For listing assets:
+  const { data: assets, isLoading: areAssetsLoading } = useXBlockAssets(usageKey);
+  const refreshAssets = useInvalidateXBlockAssets(usageKey);
+
+  // For uploading assets:
+  const handleProcessUpload = React.useCallback(async ({
+    fileData, requestConfig, handleError,
+  }: { fileData: FormData, requestConfig: any, handleError: any }) => {
+    const uploadData = new FormData();
+    const file = fileData.get('file') as File;
+    uploadData.set('content', file); // Paragon calls this 'file' but our API needs it called 'content'
+    // TODO: make the filename unique if is already exists in assets list, to avoid overwriting.
```

Comment:
I think overwriting will be a common and desired use case.

Reply:
OK, maybe a future improvement would be just to show a confirmation dialog/warning in that case.

Reply:
Comment updated ^